### ISSUE TITLE
deepFindKey is not working properly

### ISSUE THREAD

**Issue by kristiantoska** [andi7/utils#2]
![image](https://user-images.githubusercontent.com/20017227/57870841-3a9dc380-7808-11e9-803a-32de7fb2348d.png)

the implementation for checking a key's existence is too naive, the assertion should be based on the key and not the key's value!!!

**Comment by ardibegolli** [comment]
And I was thinking that `deepFindKey` was far better than lodash's `get`. Shame on me! And shame on you, too!

**Comment by kristiantoska** [comment]
@ardibegolli amen brother! And to think i doesn't even have array support!!! Even my toaster has that!!!

**Comment by ardibegolli** [comment]
yeah @kristiantoska you have a fancy toaster out there. Fancier than @andi7's lib though..
